src/main/webapp/dev/DatePickerJsTest.js

```diff
@@ -0,0 +1,23 @@
+QUnit.module('datepicker.js');
+
+QUnit.test('getMinDateForPublishDate(visibleDate)', function(assert) {
+    assert.deepEqual(getMinDateForPublishDate(new Date('10/13/2014')), new Date('10/13/2014'),
+            'returns visibleDate(parameter) itself');
+});
+
+QUnit.test('getMaxDateForVisibleDate(startDate, publishDate)', function(assert) {
+    assert.deepEqual(getMaxDateForVisibleDate(null, null), null,
+            'When startDate and publishDate are null, returns null');
+    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), null), new Date('08/25/2014'),
+            'When publishDate is null, returns startDate');
+    assert.deepEqual(getMaxDateForVisibleDate(null, new Date('08/25/2014')), null,
+            'When startDate is null, returns null');
+    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), undefined), new Date('08/25/2014'),
+            'When publishDate is undefined, returns startDate');
+    assert.deepEqual(getMaxDateForVisibleDate(new Date('10/13/2014'), new Date('08/25/2014')), new Date('08/25/2014'),
+            'When startDate > publishDate, returns publishDate');
+    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), new Date('08/25/2014')), new Date('08/25/2014'),
+            'When startDate = publishDate, returns startDate');
+    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), new Date('08/26/2014')), new Date('08/25/2014'),
+            'When startDate < publishDate, returns startDate');
+});
```

Comment:
`/**
 * Compares startDate and publishDate and returns the earlier date,
 * if publishDate is undefined or null, returns startDate as minDate
 * 
 * @assumption: startDate has a valid value
 * 
 * @param startDate
 * @param publishDate
 * 
 * @returns minDate 
 * 
 */
function getMaxDateForVisibleDate(startDate, publishDate) {
    var minDate = 0;

    if (publishDate === null || publishDate === undefined) {
        minDate = startDate;
    } else if (startDate > publishDate) {
        minDate = publishDate;
    } else {
        minDate = startDate;
    }

    return minDate;
}

/**
 * @assumption: visibleDate has a valid value
 * @returns visibleDate 
 */
function getMinDateForPublishDate(visibleDate) {
    return visibleDate;
}`

`QUnit.test('getMaxDateForVisibleDate(startDate, publishDate)', function(assert) {
    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), null), new Date('08/25/2014'),
            'When publishDate is null, returns startDate');
    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), undefined), new Date('08/25/2014'),
            'When publishDate is undefined, returns startDate');
    assert.deepEqual(getMaxDateForVisibleDate(new Date('10/13/2014'), new Date('08/25/2014')), new Date('08/25/2014'),
            'When startDate > publishDate, returns publishDate');
    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), new Date('08/25/2014')), new Date('08/25/2014'),
            'When startDate = publishDate, returns startDate');
    assert.deepEqual(getMaxDateForVisibleDate(new Date('08/25/2014'), new Date('08/26/2014')), new Date('08/25/2014'),
            'When startDate < publishDate, returns startDate');
});`

@xpdavid  i have made changes in both datepicker.js and test file please review